src/Build/Evaluation/LazyItemEvaluator.cs

```diff
@@ -342,15 +338,64 @@ private static ImmutableList<ItemData>.Builder ComputeItems(LazyItemList lazyIte
 
                 ImmutableHashSet<string> currentGlobsToIgnore = globsToIgnoreStack == null ? globsToIgnore : globsToIgnoreStack.Peek();
 
+                Dictionary<string, UpdateOperation> itemsWithNoWildcards = new Dictionary<string, UpdateOperation>(FileUtilities.GetIsFileSystemCaseSensitive() ? StringComparer.Ordinal : StringComparer.OrdinalIgnoreCase);
```

Comment:
Is this correct? Is a check like this elsewhere what causes #5749, which seems to be a regression?

Reply:
Now you mention it, I'm actually surprised this works. I put this in after [this test](https://github.com/dotnet/msbuild/blob/4f8cdfdc331b5700826be2c192b94f16034b9905/src/Build.OM.UnitTests/Definition/ProjectItem_Tests.cs#L2819-L2840) failed, but I initially misread it as saying it should be case-sensitive on linux and case-insensitive on windows, and then it apparently passed. I should add another test to make this fail, then correct this line.